### PR TITLE
This fixes Issue #72

### PR DESCRIPTION
**Pull request by pvince**
Fixed issue #72 where int32 and int64 values were only being validated as positive, unsigned values. Added validation that ensures the values are signed integers as per the OAS 3.0 specification.master

Also updated the comments to indicate these changes.

Also also added tests for various edge case values.

### CODE REVIEW

File: src/options.ts [hunk 1]
```diff
@@ -33,21 +33,29 @@ export interface ExgesisCompiledOptions {
...
-const INT_32_MAX = Math.pow(2, 32) - 1;
- // Actually 18446744073709551616-1, but Javascript doesn't handle integers this large.
-const INT_64_MAX = 18446744073709556000;
-
+const INT_32_MIN = -1 * Math.pow(2, 31);
+const INT_32_MAX = Math.pow(2, 31) - 1;
+// Javascript can only safely support a range of -(2^53 - 1) to (2^53 - 1)
+//      https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER
+const INT_64_MIN = Number.MIN_SAFE_INTEGER;
+const INT_64_MAX = Number.MAX_SAFE_INTEGER;
```
Comment (jwalton) [inline src/options.ts]: I'm not sure if it's better to make this Number.MAX_SAFE_INTEGER or Math.pow(2, 63) - 1.  If someone passes us a JSON object with 4611686018427388001, this is allowed according to the OpenAPI spec, but we are going to round it to 4611686018427388000 which is... odd.

Comment (jwalton) [reply src/options.ts]: I think we should probably allow big numbers, and let it be up to the person using the library to be aware of the fact that JavaScript is weird.  :P

**Comment by jwalton** [comment]
One other minor comment; exegesis uses semantic-release, so I can either squash-merge your commit to a 'fix: Fixes validation of int32 and int64 values.', or you can change one of your commit messages to follow this format (this gets used to generate the release notes, and to bump the version number).
